srtp/srtp.c

```diff
@@ -1006,6 +1006,10 @@ srtp_stream_init_keys(srtp_stream_ctx_t *srtp, srtp_master_key_t *master_key,
       rtp_xtn_hdr_base_key_len = base_key_length(session_keys->rtp_xtn_hdr_cipher->type,
                                                  rtp_xtn_hdr_keylen);
       rtp_xtn_hdr_salt_len = rtp_xtn_hdr_keylen - rtp_xtn_hdr_base_key_len;
+      if (rtp_xtn_hdr_salt_len > rtp_salt_len) {
```

Comment:
That's even better than what I had in mind.  I was going to take the lazy route and just make a more complex if statement, but I appreciate the switch statement that then results in an explicit error when we have a cipher mismatch and it's not due to GCM.